container/go/cmd/pusher/pusher.go

```diff
@@ -63,19 +83,29 @@ func main() {
 	if *format == "docker" && filepath.Ext(*src) != ".tar" {
 		log.Fatalf("Invalid value for argument -src for -format=docker, got %q, want path to tarball file with extension .tar.", *src)
 	}
-	if *format == "legacy" && filepath.Base(*src) != "manifest.json" {
-		log.Fatalf("Invalid value for argument -src for -format=legacy, got %q, want path to manifest.json", *src)
+	if *format == "legacy" && filepath.Base(*src) != "config.json" {
+		log.Fatalf("Invalid value for argument -src for -format=legacy, got %q, want path to config.json", *src)
 	}
 	if *format == "oci" && filepath.Base(*src) != "index.json" {
 		log.Fatalf("Invalid value for argument -src for -format=oci, got %q, want path to index.json", *src)
 	}
 	if *format == "oci" || *format == "legacy" {
 		imgSrc = filepath.Dir(*src)
+		log.Printf("Determined image source path to be %q based on -format=%q, -src=%q.", imgSrc, *format, *src)
 	}
 	if *format == "docker" {
 		imgSrc = *src
 	}
-	log.Printf("Determined image source path to be %q based on -format=%q, -src=%q.", imgSrc, *format, *src)
+	if *format == "legacy" {
+		if _, err := os.Stat(filepath.Join(imgSrc, manifestPath)); err != nil {
```

Comment:
I'm seeing `filepath.Join(imgSrc, manifestPath)` repeated in a lot of places. Do it once, assign it to a variable and use that. Same comment for the config file.